daemon/logger/logger.go

```diff
@@ -8,16 +8,22 @@
 package logger
 
 import (
-	"errors"
 	"sync"
 	"time"
 
 	"github.com/docker/docker/api/types/backend"
 	"github.com/docker/docker/pkg/jsonlog"
 )
 
-// ErrReadLogsNotSupported is returned when the logger does not support reading logs.
-var ErrReadLogsNotSupported = errors.New("configured logging driver does not support reading")
+// ErrReadLogsNotSupported is returned when the underlying log driver does not support reading
+type ErrReadLogsNotSupported struct{}
```

Comment:
Is there a reason for the different approach here than for (e.g.); `cluster.notAvailableError`?


```go
type notAvailableError string

func (e notAvailableError) Error() string {
	return string(e)
}

func (e notAvailableError) Unavailable() {}
```